### PR TITLE
[Keymap] Keychron Q1 user keymap

### PR DESCRIPTION
**Pull request by teimor**
## Description

Created a custom keymap for Keychron Q1.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [X] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/keychron/q1/rev_0100/keymaps/teimor/keymap.c [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+#include "keymap_user.h"
+#ifdef RGB_MATRIX_ENABLE
+#    include "rgb_matrix_user.h"
+#endif
+#include "tap_dance_user.c"
```
Comment (drashna) [inline keyboards/keychron/q1/rev_0100/keymaps/teimor/keymap.c:22]: c files should not be included like this. 
```suggestion
```

It should be added via `SRC += tap_dance_user.c` in the rules.mk file for your keymap.

File: keyboards/keychron/q1/rev_0100/keymaps/teimor/keymap.c [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+            return false;  // Skip all further processing of this key
+        case KC_FLXP:
+            if (record->event.pressed) {
+                register_code(KC_LWIN);
+                register_code(KC_E);
+            } else {
+                unregister_code(KC_LWIN);
+                unregister_code(KC_E);
+            }
+            return false;  // Skip all further processing of this key
```
Comment (drashna) [inline keyboards/keychron/q1/rev_0100/keymaps/teimor/keymap.c:112]: these can be replaced with `G(KC_TAB)`, and `G(KC_E)` in the keymap directly.

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/keychron/q1/rev_0100/keymaps/teimor/keymap.c (hunk 2 above)
Comment (teimor) [reply keyboards/keychron/q1/rev_0100/keymaps/teimor/keymap.c]: Fixed.

**Comment by teimor** [comment]
@drashna Thank you for the review and help, I fixed those issues. Can you review it again?

**Review by drashna — APPROVED** [review]

File: keyboards/keychron/q1/rev_0100/keymaps/teimor/tap_dance_user.c [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+#include "keymap_user.h"
+#include "tap_dance_setup.c"
```
Comment (fauxpark) [inline keyboards/keychron/q1/rev_0100/keymaps/teimor/tap_dance_user.c:19]: Don't include .c files in other .c files.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/keychron/q1/rev_0100/keymaps/teimor/tap_dance_user.c (hunk 3 above)
Comment (teimor) [reply keyboards/keychron/q1/rev_0100/keymaps/teimor/tap_dance_user.c]: @fauxpark Fixed and thank you for the review.

**Comment by teimor** [comment]
@fauxpark What will be the correct procedure? Should I request Drashna to review the changes too, or is it just you who needs to review the changes?

File: keyboards/keychron/q1/rev_0100/keymaps/teimor/README.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
+# teimor's Keychron Q1 keymap
```
Comment (drashna) [inline keyboards/keychron/q1/rev_0100/keymaps/teimor/README.md:1]: Please rename the file to be lower case.

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/keychron/q1/rev_0100/keymaps/teimor/README.md (hunk 4 above)
Comment (teimor) [reply keyboards/keychron/q1/rev_0100/keymaps/teimor/README.md]: done.

**Review by drashna — APPROVED** [review]

**Comment by teimor** [comment]
Hi @fauxpark, Sorry for tagging but I thinking this pr moved to stale again. current status is it only waiting for your review after the requested changes.

**Review by noroadsleft — APPROVED** [review]
Looks good to me.

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**